### PR TITLE
PMM-8160: Versions compatibility

### PR DESCRIPTION
**Pull request by oter**
https://jira.percona.com/browse/PMM-8160

API PR: https://github.com/percona/pmm/pull/755
managed PR: https://github.com/percona/pmm-managed/pull/785

### CODE REVIEW

**Review by Dasio — APPROVED** [review]

File: client/version.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
+// pmm-agent
```
Comment (BupycHuk) [inline client/version.go:1]: this file should be separate structure.

File: client/version.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	mysqlBin            = "mysql"
+	xtrabackupBin       = "xtrabackup"
+)
+
+var (
+	mysqlVersionRegexp      = regexp.MustCompile("^.*Ver ([!-~]*).*")
+	xtrabackupVersionRegexp = regexp.MustCompile("^xtrabackup version ([!-~]*).*")
+)
+
+func (c *Client) localMySQLVersion() (string, error) {
```
Comment (BupycHuk) [inline client/version.go:43]: `mysqlClientBinaryVersion`?

**Review by BupycHuk — CHANGES_REQUESTED** [review]

File: client/version.go (hunk 1 above)
Comment (oter) [reply client/version.go:1]: could you please add more details.

Comment (oter) [reply client/version.go:1]: could you please add more details.

File: client/version.go (hunk 2 above)
Comment (oter) [reply client/version.go:43]: I could rename it just to `mysqlVersion`. What do you think?

File: client/version.go (hunk 1 above)
Comment (BupycHuk) [reply client/version.go:1]: I mean that `localMySQLVersion`, `xtrabackupVersion` methods doesn't look like it should be part of `Client` structure, my suggestion to make it separate function like we did for `CheckConnection`

Comment (oter) [reply client/version.go:1]: I see, will do.

File: client/version.go (hunk 2 above)
Comment (BupycHuk) [reply client/version.go:43]: What do we check here? Version of MySQL client installed locally, aren't we?

Comment (oter) [reply client/version.go:43]: I need to check mysql server version instead of client  (`mysqld` instead of `mysql`) 🤦 Thanks for pointing this out!

**Pull request closed without merging**